packages/vaex-core/vaex/expression.py

```diff
@@ -460,13 +460,49 @@ def translate(id):
                 return new
         expr = expresso.translate(self.expression, translate)
         return Expression(self.ds, expr)
- 
+
     def astype(self, dtype):
         return self.ds.func.astype(self, str(dtype))
 
     def apply(self, f):
         return self.ds.apply(f, [self.expression])
 
+    def map(self, mapper):
+        '''Map values of an expression or in memory column accoring to an input
+        dictionary or a custom callable function.
+
+        :param mapper: (dict or function) used to map the values on top of the existing ones
+        :return: A vaex expression
+        :rtype: vaex.expression.Expression
+
+        Example:
+
+        >>> import vaex
+        >>> df = vaex.from_arrays(color=['red', 'red', 'blue', 'red', 'green'])
+        >>> mapper = {'red': 1, 'blue': 2, 'green': 3}
+        >>> df['color_mapped'] = df.color.map(mapper)
+        >>> df.color_mapped.values
+        array([1, 1, 2, 1, 3])
+        '''
+        if isinstance(mapper, dict):
+            def lookup_func(x):
+                if isinstance(x, np.bytes_):
+                    x = x.decode('utf-8')
+                    if x in mapper.keys():
+                        return mapper[x]
+                    else:
+                        return np.nan
+                else:
+                    if x in mapper.keys():
+                        return mapper[x]
+                    else:
+                        return np.nan
```

Comment:
we should not return nan, nan is only for numerical values. return None instead if we have to, even better would be to missing_value argument is map.

Reply:
Ok added `missing_values=None` argument. I've updated the tests, so in principle one could put `missing_values=np.nan` is that is required.